### ISSUE TITLE
Can't run call gulp-useref from parallel gulp workflows

### ISSUE THREAD

**Issue by orzarchi** [jonkemp/gulp-useref#201]
I tried to run useref from two task in parallel, using the new gulp 4 _gulp.parallel_ method. 
The result is that one of the tasks almost always gets an error (no stack),
or that I get the error **Error: stream.push() after EOF**.

After looking at index.js, the solution seem to be obvious:

```javascript
    streamManager = require('./lib/streamManager');

module.exports = function (options) {
    streamManager.transforms = Array.prototype.slice.call(arguments, 1);
    streamManager.options = options || {};

    streamManager.additionalStreams();

    return through.obj(streamManager.transformFunction, streamManager.flushFunction);
};
```
Instead of setting properties on a global streamManager object, you could just pass these in as arguments to a function.
If you approve, I'll try to find the time to submit a pull request this week.

**Comment by jonaszuberbuehler** [comment]
I _think_ I'm facing the same issue, although I'm still using gulp 3.

To support localisation (in a SEO friendly way) I process the resources in parallel streams, one for each supported locale (called version in code). My minification task looks like this:

```javascript
gulp.task('minify', function () {
    return execForLocales(function (version) {
        return gulp.src(version.tmp + '/*.html')
                .pipe(useref())
                .pipe(defer())
                .pipe(gulpif('*.html', htmlmin({
                    collapseWhitespace: true,
                    conservativeCollapse: true,
                    removeComments: false,
                    minifyCSS: true,
                    minifyJS: true
                })))
                .pipe(gulpif('*.js', uglify()))
                .pipe(gulpif('*.css', cssnano()))
                .pipe(gulp.dest(version.dist));
    });
});

function execForLocales(task) {
    var tasks = versions.map(function (version) {
        return task(version);
    });
    return merge(tasks);
}
```

This all works if I process it for one locale only but fails without any error if I try to process more locales.

**Comment by orzarchi** [comment]
Yup seems similar alright, me using gulp4 probably has nothing to do with
it.

**Comment by jonkemp** [comment]
This looks like a bug report. In order to be as helpful as possible, here are some guidelines for bug reports:

1. **Use the GitHub issue search** &mdash; check if the issue has already been reported.

2. **Check if the issue has been fixed** &mdash; try to reproduce it using the latest `master` or development branch in the repository.

3. **Isolate the problem** &mdash; create a [reduced test case](http://css-tricks.com/6263-reduced-test-cases/) and a live example.

A good bug report shouldn't leave others needing to chase you up for more information. Please try to be as detailed as possible in your report. What is your environment? What steps will reproduce the issue? What browser(s) and OS experience the problem? What would you expect to be the outcome? All these details will help people to fix any potential bugs. Thanks!

**Comment by bryk** [comment]
+1 - this is also a problem for our parallel build pipeline

**Comment by orzarchi** [comment]
Still hoping for @jonkemp to weigh in on this.. 
In the mean time I've switched to gulp-usemin, as that seems to work with parallel workflows.

**Comment by jonkemp** [comment]
Have you tried it with the latest version 3.1.0?

**Comment by orzarchi** [comment]
Yep, still have the same problems.
I'll try to create a minimal reproduction, but really you just need to run any sort of build in parallel.

**Comment by jonkemp** [comment]
I am working on a refactor which would get rid of streamManager.js. I plan on merging it in sometime this weekend. That may do what you are wanting.

**Comment by noderat** [comment]
Yeah, looks like reverting to 3.0.4 has resolved this issue for now, until @jonkemp merges in his changes to remove streamManager

**Comment by jonkemp** [comment]
Version 3.1.2 is out. Please try with the latest version and see if that fixes the issue.

**Comment by danjarvis** [comment]
Just hit the same issue running 3.1.2

**Comment by NullSoldier** [comment]
Same issue running 3.1.2 except it's june of 2017 now......

**Comment by jonkemp** [comment]
@NullSoldier let me know if you're willing to help!
